contracts/DomainRegister.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.21;
+
+contract DomainRegister {
+    uint public  minimalPledge;
+
+    mapping (string => address) public register;
+    mapping (string => uint) domainPledge;
```

Comment:
You could map `string` to some `struct` instead of these two maps. This is better in terms of future extension of functionality and may save you gas in some cases.
For example like this below
```
        delete register[_domain];
        delete domainPledge[_domain];
```
two key-value resolutions instead of one

-1 point